src/config/server.rs

```diff
@@ -17,8 +17,9 @@ pub struct Server {
   pub enable_response_validation: Option<bool>,
   pub global_response_timeout: Option<i64>,
   pub port: Option<u16>,
-  pub proxy: Option<Proxy>,
   pub vars: Option<BTreeMap<String, String>>,
+  #[serde(default, skip_serializing_if = "Option::is_none")]
+  pub upstream: Option<Upstream>,
```

Comment:
now as we marked the property with serde(default) we can remove wrapping with `Option` and leave the type itself. The serde will put the default value in that case if user isn't define it.

Just make sure that `skip_serializing_if` will equal to `is_default` in that case (see examples in other places in repo)

Reply:
server operator is not required so changing test undesirable. To remove wrapping and to be able to leave tests as is try to use `skip_serializing_if` option as I've mentioned. Check [the example](https://github.com/tailcallhq/tailcall/blob/458716df62eacd670e8ccd20f9c10160ff752bc4/src/config/config.rs#L211) of it

Reply:
@akhill10 I think this is a different problem. `Eq` is not implemented.

Reply:
Thanks, fixed.